src/hooks/useAppState/reducer.js

```diff
@@ -1,23 +1,30 @@
 export const types = {
-  SET_LIST_ID: 'SET_LIST_ID',
+  NEW_LIST: 'NEW_LIST',
   SET_STATE: 'SET_STATE',
   NEW_ITEM: 'NEW_ITEM',
   DELETE_ITEM_BY_KEY: 'DELETE_ITEM_BY_KEY',
   UPDATE_ITEM_BY_KEY: 'UPDATE_ITEM_BY_KEY',
   TOGGLE_COMPLETION_BY_KEY: 'TOGGLE_COMPLETION_BY_KEY',
   STORE_ID: 'STORE_ID',
+  STORE_LIST_ID: 'STORE_LIST_ID',
   MERGE_ITEMS: 'MERGE_ITEMS',
 };
 
-export const initialState = { items: [], listId: null };
+export const initialState = { items: [], lists: [], index: 0 };
 
 const reducer = (state, { type, data }) => {
   switch (type) {
-    case types.SET_LIST_ID:
-      return { ...state, listId: data.listId };
+    case types.NEW_LIST: {
+      const { key, name } = data;
+      return {
+        ...state,
+        items: [...state.items, []],
+        lists: [...state.lists, { key, name }],
```

Comment:
Both lists and items grow by one, the new array in items is empty for now (since we have no items in the new list).